contracts/src/AgentExecutor.sol

```diff
@@ -38,4 +43,9 @@ contract AgentExecutor {
     function _transferToken(address token, address recipient, uint128 amount) internal {
         IERC20(token).safeTransfer(recipient, amount);
     }
+
+    /// @dev Call a contract at the given address, with provided bytes as payload.
+    function _executeCall(address target, bytes memory payload, uint256 dynamicGas) internal {
+       target.call{gas: dynamicGas}(payload);
```

Comment:
Should check gas left and handle error here.

Reply:
Yes, we need to check result and revert if error happens, something like 
https://github.com/axelarnetwork/axelar-cgp-solidity/blob/main/contracts/util/Caller.sol